### PR TITLE
test: account pages

### PR DESCRIPTION
**Pull request by sshanzel**
### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-389 #done

### CODE REVIEW

**Comment by sshanzel** [comment]
One of the awesome things about creating test cases is the opportunity to refactor the code. I centralized some calls to avoid unnecessary requests. Also fixed session verification process for linking accounts.

With that, I will branch out from this the code required for validating the linking/unlinking of social providers.

cc @rebelchris would love to hear what you think 🚢

File: packages/webapp/__tests__/AccountSecurityPage.tsx [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  expect(el).toHaveValue(defaultLoggedUser.email);
+  const displayForm = await screen.findByText('Change email');
+  fireEvent.click(displayForm);
+  const email = 'sample@email.com';
+  fireEvent.input(screen.getByPlaceholderText('Email'), {
+    target: { value: email },
+  });
+  const { nodes } = settingsFlowMockData.ui;
+  const token = getNodeValue('csrf_token', nodes);
+  const params = {
```
Comment (rebelchris) [inline packages/webapp/__tests__/AccountSecurityPage.tsx:163]: We'll have to catch this flow now with the new (non-required) fields

**Review by rebelchris — CHANGES_REQUESTED** [review]
Wow impressive work overall!
Some minor comments, really love the refactors here

File: packages/shared/__tests__/fixture/auth.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
 import nock from 'nock';
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/shared/__tests__/fixture/auth.ts:1]: These fixtures are used on other test suites (test suites in the shared components and the webapp) as well that's why I kept them here. For when a test suite requires any of the flow we have, we can use these data. WDYT?

[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/__tests__/fixture/auth.ts:1]: Yeah the mocks are fine, thought I saw actual tests there as well, but don't see them now, so good to go.

File: packages/webapp/__tests__/AccountOthersPage.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      isFallback: false,
+    };
+  },
+}));
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const defaultLoggedUser: LoggedUser = {
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/webapp/__tests__/AccountOthersPage.tsx:29]: Totally right! Let me refactor these ones.

File: packages/webapp/__tests__/AccountSecurityPage.tsx [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+      isFallback: false,
+    };
+  },
+}));
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const defaultLoggedUser: LoggedUser = {
```
[the remark this answers is not in the document]
Comment (sshanzel) [reply packages/webapp/__tests__/AccountSecurityPage.tsx:42]: Can you check if the changes I made were acceptable? Or should I make changes to the user fixture to have the data I needed?

**Review by rebelchris — APPROVED** [review]
I'm happy with this, but it's up to you we can also move the fields to the defaultUser, just means we have to adjust other snapshots.

**Comment by sshanzel** [comment]
> I'm happy with this, but it's up to you we can also move the fields to the defaultUser, just means we have to adjust other snapshots.

I think I will keep this change for now since the tests are specific to those fields. Thank you for the review! 🚢

**Pull request merged**